api/v1/db/aws/aws-operations.js

```diff
@@ -55,26 +53,126 @@ const objectExists = (key, bucket = thisBucket) => new Promise((resolve, reject)
   });
 });
 
+/**
+ * @summary Gets metadata on an object by making a head-object request
+ * @function
+ * @param {string} key Key of the object
+ * @param {string} bucket Bucket where the object exists
+ * @returns {Promise} Promise object representing the response
+ */
+const headObject = (key, bucket = thisBucket) => {
+  const params = { Bucket: bucket, Key: key };
+  return s3.headObject(params).promise();
+};
+
+/**
+ * @summary List objects in a bucket
+ * @function
+ * @param {Object} params Additional params to be used in the search
+ * @param {string} bucket The bucket to search for objects
+ * @returns {Promise} Promise object representing the objects
+ */
+const listObjects = (params = {}, bucket = thisBucket) => {
+  const newParams = Object.assign({ Bucket: bucket }, params);
+  return s3.listObjectsV2(newParams).promise();
+};
+
 /**
  * @summary Gets an object from a bucket
  * @function
- * @param key The key of the object
- * @param bucket The bucket where the object exists
+ * @param {string} key The key of the object
+ * @param {string} bucket The bucket where the object exists
  * @returns {Promise} Promise object representing the object response
  */
-const getObject = (key, bucket = thisBucket) => new Promise((resolve, reject) => {
+const getObject = (key, bucket = thisBucket) => {
   const params = { Bucket: bucket, Key: key };
-  // TODO: use valid credentials
-  s3.getObject(params).promise().then((data) => {
+  return s3.getObject(params).promise();
+};
+
+/**
+ * @summary Uploads a new directory object to a bucket
+ * @function
+ * @param {string} key The key of the object. Must end with "/"
+ * @param {Object} params Additional params to be used when creating the directory
+ * @param {string} bucket The bucket that the object will be uploaded to
+ * @returns {Promise} Promise object representing the response
+ */
+const putDir = (key, params = {}, bucket = thisBucket) => {
+  if (key[key.length - 1] !== '/') {
```

Comment:
Since we have installed lodash, I believe you can use [`_.last()`](https://lodash.com/docs/#last) to get the last element from an array/ a string.

Reply:
Nice, I didn't know about this method